src/handlers/web3.ts

```diff
@@ -395,24 +397,24 @@ export async function estimateGasWithPadding(
       logger.info('⛽ returning orginal gas estimation', {
         esimatedGas: estimatedGas.toString(),
       });
-      return estimatedGas.toString();
+      estimatedGasLimit = estimatedGas.toString();
     }
     // If the estimation is below the last block gas limit, use the padded estimate
     if (greaterThan(lastBlockGasLimit, paddedGas)) {
       logger.info('⛽ returning padded gas estimation', { paddedGas });
-      return paddedGas;
+      estimatedGasLimit = paddedGas;
     }
     // otherwise default to the last block gas limit
     logger.info('⛽ returning last block gas limit', { lastBlockGasLimit });
-    return lastBlockGasLimit;
+    return estimatedGasLimit;
   } catch (e: any) {
     /*
      * Reported ~400x per day, but if it's not actionable it might as well be a warning.
      */
     logger.warn('Error calculating gas limit with padding', {
       message: e.message,
     });
-    return null;
```

Comment:
before this would return null, we not return whatever value we were able to calculate before erroring